pypit/armlsd.py

```diff
@@ -129,39 +133,42 @@ def ARMLSD(fitsdict, reuseMaster=False, reloadMaster=True):
             if ('trace'+settings.argflag['reduce']['masters']['setup'] not in settings.argflag['reduce']['masters']['loaded']):
                 ###############
                 # Determine the edges of the spectrum (spatial)
-                #lordloc, rordloc, extord = artrace.trace_slits(slf, slf._mstrace[det-1], det, pcadesc="PCA trace of the slit edges")
-                lordloc, rordloc, extord = artrace.driver_trace_slits(slf._mstrace[det-1],
-                                                                      slf._pixlocn[det-1],
-                                                                      det=det,
-                                                                      settings=settings.argflag,
-                                                                      binbpx=slf._bpix[det-1],
-                                                                      armlsd=True)
-                slf.SetFrame(slf._lordloc, lordloc, det)
-                slf.SetFrame(slf._rordloc, rordloc, det)
+                #lordloc, rordloc, extord = artrace.driver_trace_slits(slf._mstrace[det-1],
+                #                                                      slf._pixlocn[det-1],
+                #                                                      det=det,
+                #                                                      settings=settings.argflag,
+                #                                                      binbpx=slf._bpix[det-1],
+                #                                                      armlsd=True)
+                Tslits = traceslits.TraceSlits(slf._mstrace[det-1], slf._pixlocn[det-1],
+                                               det=det, settings=settings.argflag, binbpx=slf._bpix[det-1])
+                tslits_dict = Tslits.run(armlsd=True)
+
+                # Save in slf
+                slf.SetFrame(slf._lordloc, tslits_dict['lcen'], det)
```

Comment:
I haven't looked at the `TraceSlits` class yet, but is, e.g., `lcen` kept as an attribute of `TraceSlits`?  Maybe you're just trying to keep things separate, but why do we need the `tslits_dict` object?  I think it's cleaner if you have, e.g., `Tslits.lcen` that then gets passed in this call to `SetFrame`.

Reply:
I agree with you.  I am forseeing that we will be carrying
around a small object outside of the 'slf' that will contain
the slit info.  And in trying to hold the line about not passing
any non-base object (e.g. dict, array) to any other function
I decided to build this dict.

But, we might well just pass Tslit.

Am going to remove the dict part for now.  @jhennawi will 
be able to handle it.  ;)

Reply:
I'm fine with having objects passed into routines provided they are providing value and not adding confusion. I think that the trace slits class is a good example of an object adding value, and indeed we had tset_slits structures in IDL which contained the same information. What I disagree is fancy object oriented stuff where it is not necessary. The arflux module is a good example of this object oriented idea gone wrong. It turns out to be a major struggle to flux the data outside the regular running of PYPIT, which is what a user is inclined to do if they want to change their sensitivity function. But in that case, the driver routine is written with the self object everywhere and relies on being called with PYPIT making the code effectively useless outside of PYPIT.

Reply:
ps.  Am keeping it internally (i.e. the fill_trace_slit_dict method).
Am just not using it anymore.